### PR TITLE
Forms - list views - filters - user roles - and more

### PR DESCRIPTION
**Pull request by 2ynn**
I pushed the develop branch to https://saskatoon-dev.lesfruitsdefendus.org

I would appreciate if you could test it, especially all the different forms:
- [ ] new owner based on pending_property (see the "register" button next to "contact" title in a pending property detail page)
- [ ] new person (pickleader, core-member, volunteer, etc..) 
- [ ] new (organization + contact person)
- [ ] new (property + owner)

I the meantime I will work on a document to list all the new features to test so the core members can try it too.

### CODE REVIEW

**Comment by tristanlatr** [comment]
Both my accounts do not work on the dev server, they was working before, though. 

I've tried the reset password form, but did not receive an email. 

Could you please send me my new password ?

Thanks again @2ynn !

**Comment by 2ynn** [comment]
For now I haven't merged the large database on the dev server and am still only using the fixtures for testing. Credentials are stored in LFD's 1Password account.
I sent you a separate email.

Cheers,

qynn



Sep 30, 2021, 20:33 by ***@***.***:

**Comment by tristanlatr** [comment]
Hello @2ynn,

I've been looking at the dev instance of saskatoon. First of all, big up to you! It's really a great job. I'm still testing the use cases you mentioned and I should get back to you with a complete feedback soon. 

Sorry for the delay.

**Comment by 2ynn** [comment]
qynn



Oct. 30, 2021, 06:21 by ***@***.***:

**Comment by tristanlatr** [comment]
Oh... Sorry I should have done this earlier...  

I've spotted a few bugs. Let a couple minutes to wrap it out.

**Comment by tristanlatr** [comment]
First of all, I want to say that I'm extremely grateful to you, Qynn, thanks a lot for the work that you've put into Saskatoon.

Here are my notes: 

- General notes
    - Having a favicon with our logo would be nice
    - The statistic panel has not beed ported to the new version, we use it to populate our annual review, so it would be good to have it (out of topic?).
    - I haven’t tested whether the server is sending emails when a picker is accepted, etc. What are all the cases in which the server is sending email BTW ?

- Login page: 
	- No need to have the red banner on the login page since it’s designed for pick leaders only. Also it says “Please read, complete and return to…” but the docs to complete aren’t linked. This was introduced in order to have clear COVID protocol, the core members should be asked if it will still be applicable for season 2022. 

- New harvest form:
    - Little frustration when selecting a date under Chrome : the already entered values are displayed in the dropdown list that is over the calendar box, used to actually select the right date (we need to press espace to remove that dropdown list). I think it’s as simple as adding ``autocomplete="off”`` to the input tag. This issue is also present on other calendar inputs. 

- Harvest view:
    - Graphical view (histogram) on the right of the "Pickers requests" or “Total harvested” is not representative of anything special. I suggest we remote it because it can confuse users. 
    - Fruit distribution:
        - When no picker has been explicitly accepted yet, there is still a picker listed in the list of recipient : “Volunteer Picker”. When selecting this person and clicking “Save”, it triggers a server error 500. Actually even when selecting a valid picker, an organization or the tree owner, the server crashes with an error 500.

- “Request to join this harvest” form
    - invalid email “invalid@email” get the server to respond with an error 500 instead of kindly refusing the output. 
    - Invalid phone number “123” get accepted like it was valid. 

- “Add a new property” (public) 
    - After validation, get us to the page ”property/create_public/thanks" which shows an error Not Found
    - The phone number is not validated
    - The email address is correctly validated, whereas in the “Request to join this harvest”, the error message should be displayed in red, the CSS class is “invalid-feedback”. 
    - Postal code  is not validated, the form should kindly fail if an invalid Canada postal code is provided

- Property view
    -  Pending property “Register” form: 
        - When we enter an invalid email like “g.brassens@myidol”, it does not save the entered values precedently (if overridden by core member) and do not add the invalid feedback box saying the email is invalid, it only reload the form with the values populated from the public form. 
        - The phone number is not validated
        - A proposition: If the property feature "Has already been registered in the past" is true, the register form could start with an dropdown list saying “This owner should be already in our database, you can select them here“. The thing to keep in mind is that the tree owner might want to update it’s information (i.e. phone number) from one year to another, so the form should allow a sort of comparison between the already in database person and the “create_public” form infos. I know it’s not a clear proposition, it’s probably something to talk about with Simone, also.

- New User form
    - When entering invalid data (like invalid email “g.brassens@myidol”), the form is simply reset, and the filled data is lost. I suppose the form should keep the entered data and add the invalid feedback on the bottom of the invalid field
    - Postal code is not validated
    - Phone number is not validated
    - A link to the admin panel setting/resetting the password of a person should be added to the “Person Update” form for better usability.

File: INSTALL.md [hunk 1]
```diff
@@ -31,7 +31,20 @@ Please follow each part of this documentation in order to run your own instance
...
-    NB: you might run into issues when installing the `Pillow` package using `pip`, in which case installing the following dependencies should help:
+
+3. Redis server
+
+    `django-redis` is currently used as a caching backend with default configuration (see `CACHES` variable in `saskatoon/settings.py`). A Redis server must be run in the background:
+    ```
+    $ sudo apt install redis-server
+    $ sudo systemctl status redis-server
+    ```
+    Note the Redis service will start automatically when the installation finishes (if using systemd):
```
Comment (tristanlatr) [inline INSTALL.md:42]: Is this still applicable?

**Comment by 2ynn** [comment]
Thanks @tristanlatr  for all this!

1. General notes
 - [ ] TODO: add favicon
 - statistics: yes out of scope for this merge (see #115). Note I have hidden the statistics tab for now.
 -  I haven't tested or changed the email process except avoiding errors when no email backend is set (necessary for fixtures).  I think all email related actions are located in `harvest/signals.py`
 -  I will double check permissions. This still need to be clarified (see #104)

2. login page
 - [ ] TODO update document link in red banner (only for logged in pickleaders/core-members)

3. Harvest form
- [ ] TODO test autocomplete="off" for all date pickers (Chrome)

4. Harvest view
 - [ ] TODO remove histrogram decoration
 - [ ] TODO Fix Fruit Distribution Form (Recipient list + Server error 500) wtf?!

5.  Form Validation
- implement form validations (see #116). It's something I hadn't looked into, I simply rearranged existing forms.

6. public property form
- fix thanks template: yes known issue, will fix along with #90

7. Property View
- Yes Issue #71 is still applicable. I didn't necessarily want to implement the "compare 2 forms side-by-side" functionality because ideally (eventually) we would like to go with a OneTimePassword sent to owners so they can edit their own forms. (before this is possible roles and permissions need to be nailed and tested, including assigning proper roles to all existing users).
- [ ] TODO make use of the `has_registered_before` field to clearly indicate if a pending property needs to be compared against another one in the database. I could eventually start the process of searching for an existing property based on email and present a link to be open in another tab.

8. Change password
The register button (+) in the Community list is meant for core-members to set the initial password for users after adding them (if applicable). 
If you click on your username (or the face icon) there is a change_password link to change your own password.
Also we have a "Forgot password" functionality on the login page (the question mark) which sends a new password by email for registered user.
So technically core-members shouldn't have to reset passwords for others very often. I'm not sure if having a link in the form is necessary. (just want to avoid having to check permissions in the form if possible). I could always add a reset password link in the community list for core-members (when user is registered, instead of the (+) it could be another icon that links to reset_password
- [ ] TODO add reset password link in community list (for core-members only)

I will try and get all the todos done for this merge. The other issues I have mentioned could probably be dealt separately.

File: INSTALL.md (hunk 1 above)
Comment (2ynn) [reply INSTALL.md:42]: Yes this still applies (see #55)

**Comment by 2ynn** [comment]
After discussion/tests w/ Simone:
TODO
- [ ]  replace "None" by pending_contact_name in Property list
- [ ] pickleader should be able to change a property (see doc/permissions.pdf)
- [ ] `+new` button in Harvest detail should be bigger (+New Harvest)
- [ ] fix search query in Autocompletes
- [ ] add core-members to pickleader-autocomplete
- [ ]  Pickleader autocomplete in Harvest Form behaves strangely (pk / auth_user.id mixup ?)
- [ ] set role=volunteer for newly created volunteers

Nice to have:
- [ ] add a pencil in Harvest list to jump directly to Edit form
- [ ] add a quick Harvest Status change form
- [ ] Basic Web page model for landing page (see #117)

**Comment by tristanlatr** [comment]
Hello, 

Does the latest changes are available on the saskatoon-dev.lesfruitsdefendus.org instance ?

Thanks

**Comment by 2ynn** [comment]
@tristanlatr just pushed develop to the dev server. You can always check the prod branch to see where the dev server is at.

**Comment by tristanlatr** [comment]
> Please let me know if you think we need a different logo (e.g. the same as in the header?)

Yes I think the same icon would be more appropriate. 

Thanks

**Comment by 2ynn** [comment]
@tristanlatr is it okay if I still push to develop directly until we merge this PR?

**Comment by tristanlatr** [comment]
Yes please go ahead. But let’s merge this PR I the coming days :)

**Review by tristanlatr — APPROVED** [review]
Let's do it!

Special thanks to @2ynn for all the the volunteer work!

**Comment by mina-andajos** [comment]
Do we continue submitting PRs to the `develop` branch or should we switch to master?

**Comment by tristanlatr** [comment]
You can target master now!

**Pull request merged**